examples/templates/clusterclass.yaml

```diff
@@ -0,0 +1,459 @@
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3ClusterTemplate
+metadata:
+  name: example-cluster-template
+spec:
+  template:
+    spec:
+      controlPlaneEndpoint:
+        host: 127.0.0.1
+        port: 6443
+      noCloudProvider: true
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlaneTemplate
+metadata:
+  name: example-controlplane
+spec:
+  template:
+    spec:
+      kubeadmConfigSpec:
+        clusterConfiguration:
+          apiServer:
+            extraArgs:
+              cloud-provider: baremetal
+          controllerManager:
+            extraArgs:
+              cloud-provider: baremetal
+        initConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: baremetal
+            name: '{{ ds.meta_data.local_hostname }}'
+        joinConfiguration:
+          controlPlane: {}
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: baremetal
+            name: '{{ ds.meta_data.local_hostname }}'
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3MachineTemplate
+metadata:
+  name: example-md-0
+spec:
+  nodeReuse: false
+  template:
+    spec:
+      automatedCleaningMode: metadata
+      image:
+        checksum: ${IMAGE_CHECKSUM}
+        checksumType: ${IMAGE_CHECKSUM_TYPE}
+        format: ${IMAGE_FORMAT}
+        url: http://example.com
+      dataTemplate:
+        name: example-md-metadata
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3DataTemplate
+metadata:
+  name: example-md-metadata
+spec:
+  clusterName: example
```

Comment:
Too bad we have the clusterName on the Metal3DataTemplate and IPPool :disappointed: 
I guess this makes it quite hard to use them from the ClusterClass? Would it make more sense to have them separate or in the cluster.yaml?